src/xmagics.hpp

```diff
@@ -0,0 +1,251 @@
+            else
+                order = 3;
+            output.precision(precision);
+            output << timespan * scaling[order] << " " << units[order];
+            return output.str();
+        }
+
+        void execute(std::string & options, std::string & code) const
+        {
+            auto opts = parse_opts(options, "n:r:p:qo");
+
+            std::size_t number = (opts.find("n") != opts.end())? std::atoi(opts["n"].c_str()): 0ul;
+            std::size_t default_repeat = 7;
+            std::size_t repeat = (opts.find("r") != opts.end())? std::atoi(opts["r"].c_str()): default_repeat;
+            std::size_t precision = (opts.find("p") != opts.end())? std::atoi(opts["p"].c_str()): 3;
+            bool quiet = (opts.find("q") != opts.end());
+            bool return_result = (opts.find("o") != opts.end());
+
+            if (trim(code).empty())
+                return;
+
+            cling::Value result;
+            cling::Interpreter::CompilationResult compilation_result;
+
+            auto errorlevel = 0;
+            try
+            {
+                if (number == 0)
+                    for(std::size_t n=0; n<10; ++n)
+                    {
+                        number = std::pow(10, n);
+                        std::string timeit_code = inner(number, code);
+                        errorlevel = m_processor->process(timeit_code.c_str(), compilation_result, &result);
+                        if (result.simplisticCastAs<double>() >= 0.2)
+                            break;
+                    }
+
+                std::vector<double> all_runs;
+                double mean = 0;
+                double stdev = 0;
+                for(std::size_t r=0; r<repeat; ++r)
+                {
+                    std::string timeit_code = inner(number, code);
+                    errorlevel = m_processor->process(timeit_code.c_str(), compilation_result, &result);
+                    all_runs.push_back(result.simplisticCastAs<double>()/number);
+                    mean += all_runs.back();
+                }
+                mean /= repeat;
+                for(std::size_t r=0; r<repeat; ++r)
+                    stdev += (all_runs[r] - mean)*(all_runs[r] - mean);
+                stdev = std::sqrt(stdev/repeat);
+                auto minmax = std::minmax_element(all_runs.begin(), all_runs.end());
+                std::cout << _format_time(mean, precision) << " +- " << _format_time(stdev, precision);
+                std::cout << " per loop (mean +- std. dev. of " << repeat << " run" << ((repeat==1)? ", ":"s ");
+                std::cout << number << " loop" << ((number==1)? "":"s") << " each)\n";             
+            }
+            catch (cling::InterpreterException& e)
+            {
+                if (!e.diagnose())
+                {
+                    std::cerr << "Caught an interpreter exception!\n"
+                           << e.what() << '\n';
+                }
```

Comment:
I think we should notify the user even if `e.diagnose()` it `true`, else we may have a silent exception.